### ISSUE TITLE
make[1]: *** [interpreter/llvm/src/utils/TableGen/CMakeFiles/llvm-tblgen.dir/all] Error 2

### ISSUE THREAD

**Issue by nmik** [cositools/cosi-setup#3]
Hello,
I run the installation od the COSItools on my MacBook Pro (M1 Chip, MacOS Monterey, Hombrew) with the command:

`/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/cositools/cosi-setup/feature/initialsetup/setup.sh)" --extras=cosi-data-challenge-1`

The installation fails at the ROOt installation stage (BuildLogROOT.txt is attached). The error seems to be related to clang but I wasn't able to understand it fully:

ld: entry point (_main) undefined. for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [interpreter/llvm/src/bin/llvm-tblgen] Error 1
make[1]: *** [interpreter/llvm/src/utils/TableGen/CMakeFiles/llvm-tblgen.dir/all] Error 2

[BuildLogROOT.txt](https://github.com/cositools/cosi-setup/files/10744768/BuildLogROOT.txt)

**Comment by zoglauer** [comment]
First, the call should be:
```
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/cositools/cosi-setup/main/setup.sh)" _ --extras=cosi-data-challenge-1
```
You missed the underscore. But that should not cause your error.

I think I saw that error before and at that time is was a transient error probably due to compiling on multiple cores, i.e., it went away when I tried again. Thus could you just try again? If it fails again, use --maxthreads=1

**Comment by nmik** [comment]
Thank you,
I rebooted my laptop and then run first 
`/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/cositools/cosi-setup/main/setup.sh)" _ --extras=cosi-data-challenge-1 `
which didn't work, and then
`/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/cositools/cosi-setup/main/setup.sh)" _ --extras=cosi-data-challenge-1 --maxthreads=1`.
Unfortunately I get the same error as before both times. The new BuildLogROOT.txt is attached.



[BuildLogROOT.txt](https://github.com/cositools/cosi-setup/files/10759150/BuildLogROOT.txt)

**Comment by nmik** [comment]
Thanks to the help of Sean Pike, I was able to solve this issue. It was related to homebrew installation which I inherited from my previous mac backup, which had an intel chip instead of M1 chip. This can be checked by looking for the location of the homebrew package:

`>>> which brew`
`/usr/local/bin/brew`

This means it is the intel version. While

`>>> which brew`
`/opt/homebrew/bin/brew`

is the version for M1.

To uninstall hombrew for intel:
`/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/uninstall.sh)"`

To install the hombrew for M1:
`/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"`

I should mention that I also:

- Updated the OSX to Ventura
- Uninstalled and reinstalled Xcode and CommandLineTools
- be sure to have the CommandLineTools in the right place:
`sudo xcode-select -s /Applications/Xcode.app/Contents/Developer`
